### PR TITLE
Schema: include consistent IDs, start time and end time in flapping and acknowledgement history

### LINKED ISSUE #81: History: Only insert one Flapping & Acknowledgement row
We currently insert one row for start/set and one row for stop/clear. The first row could be updated/replaced using the timestamp.

Depends on https://github.com/Icinga/icinga2/issues/7689

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #81

### CODE REVIEW

**Comment by N-o-X** [comment]
@Al2Klimov can't assign you for review, please just review and leave a comment.

File: configobject/history/history.go [hunk 1]
```diff
@@ -457,27 +457,38 @@ func flappingHistoryWorker(super *supervisor.Supervisor) {
...
+			`is_persistent=COALESCE(CASE WHEN is_persistent = 'n' THEN null ELSE is_persistent END, VALUES(is_persistent))`,
 		`REPLACE INTO history (id, environment_id, endpoint_id, object_type, host_id, service_id,` +
 			` acknowledgement_history_id, event_type, event_time)` +
 			`VALUES (?,?,?,?,?,?,?,?,?)`,
 	}
 
 	dataFunctions := []func(values map[string]interface{}) []interface{}{
 		func(values map[string]interface{}) []interface{} {
-			id := uuid.MustParse(values["id"].(string))
+			log.Info(values["event_type"].(string));
```
Comment (Al2Klimov) [inline configobject/history/history.go]: What's this for?

Comment (N-o-X) [reply configobject/history/history.go]: removed...

**Comment by N-o-X** [comment]
@Al2Klimov review again please.

File: configobject/history/history.go [hunk 2]
```diff
@@ -452,27 +457,37 @@ func flappingHistoryWorker(super *supervisor.Supervisor) {
...
+			`VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?)` +
+			`ON DUPLICATE KEY UPDATE ` +
+			`set_time=IFNULL(NULLIF(set_time, 0), VALUES(set_time)),` +
+			`clear_time=IFNULL(NULLIF(clear_time, 0), VALUES(clear_time)),` +
+			`author=IFNULL(NULLIF(author, ''), VALUES(author)),` +
+			`cleared_by=IFNULL(NULLIF(cleared_by, ''), VALUES(cleared_by)),` +
+			`comment=IFNULL(NULLIF(comment, ''), VALUES(comment)),` +
+			`expire_time=IFNULL(NULLIF(expire_time, 0), VALUES(expire_time)),` +
+			`is_sticky=IFNULL(NULLIF(is_sticky, 'n'), VALUES(is_sticky)),` +
+			`is_persistent=IFNULL(NULLIF(is_persistent, 'n'), VALUES(is_persistent))`,
```
Comment (Al2Klimov) [inline configobject/history/history.go]: Are you sure about NULLIF for bools?

File: configobject/history/history.go [hunk 3]
```diff
@@ -481,18 +496,26 @@ func acknowledgementHistoryWorker(super *supervisor.Supervisor) {
 			return data
 		},
 		func(values map[string]interface{}) []interface{} {
-			id := uuid.MustParse(values["id"].(string))
+			eventId := uuid.MustParse(values["event_id"].(string))
```
Comment (Al2Klimov) [inline configobject/history/history.go]: Don't you want to inline this one like id?

File: configobject/history/history.go [hunk 4]
```diff
@@ -582,3 +605,11 @@ func removeEntryFromEntriesSlice(s []redis.XMessage, i int) []redis.XMessage {
 	s[len(s)-1], s[i] = s[i], s[len(s)-1]
 	return s[:len(s)-1]
 }
+
+func defaultIfNil(value interface{}, defaultValue interface{}) interface{} {
```
Comment (Al2Klimov) [inline configobject/history/history.go]: * Docs would be helpful
* You don't need to specify the same param type twice in a row

File: configobject/history/history.go (hunk 3 above)
Comment (N-o-X) [reply configobject/history/history.go]: I would like to, but you can't.

**Comment by N-o-X** [comment]
@Al2Klimov review again please 😄

**Comment by N-o-X** [comment]
Ready to be merged, waiting for https://github.com/Icinga/icinga2/pull/7690

**Pull request merged**